### PR TITLE
Bump languageVersion/apiVersion to Kotlin 2.0

### PR DESCRIPTION
**Pull request by BoD**
Also, I tried to enable KSP 2 but it crashes due to https://github.com/google/ksp/issues/1823 (which needs a fix that will be in Kotlin 2.1).

### CODE REVIEW

File: libraries/apollo-debug-server/build.gradle.kts [hunk 1]
```diff
@@ -123,3 +126,11 @@ dependencies {
     to.attribute(artifactType, ArtifactTypeDefinition.DIRECTORY_TYPE)
   }
 }
+
+// Can't use apiVersion KOTLIN_2_0 when using languageVersion KOTLIN_1_9, which is the case here because we're using KSP 1
```
Comment (martinbonnin) [inline libraries/apollo-debug-server/build.gradle.kts:130]: Where is `languageVersion` set to `KOTLIN_1_9`?  Is KSP overriding our default?

Comment (BoD) [reply libraries/apollo-debug-server/build.gradle.kts:130]: I'm not actually sure, but guessing yes, KSP must automatically switch to K1. Interestingly, if I set  `languageVersion.set(KotlinVersion.KOTLIN_1_9)` here, I'm getting an error:
```
Inconsistent settings for Kotlin source sets: 'jvmMain' depends on 'jvmCommonMain'
  'jvmMain': language version is KOTLIN_1_9
  'jvmCommonMain': language version is KOTLIN_2_0
  The language version of the dependent source set must be greater than or equal to that of its dependency.
```

File: build-logic/src/main/kotlin/api.kt [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 import org.jetbrains.kotlin.gradle.targets.js.npm.tasks.KotlinNpmInstallTask
 
 class AndroidOptions(
-    val withCompose: Boolean
+    val withCompose: Boolean,
+)
+
+class KotlinCompilerOptions(
+    val apiVersion: KotlinVersion = KotlinVersion.KOTLIN_2_0,
+    val languageVersion: KotlinVersion = KotlinVersion.KOTLIN_2_0,
```
Comment (martinbonnin) [inline build-logic/src/main/kotlin/api.kt:17]: Nit: is there any occurrence where we're not setting the same value for both parameters? I would probably have gone with a single parameter but 2 work too 👍

**Review by martinbonnin — APPROVED** [review]

File: build-logic/src/main/kotlin/api.kt (hunk 2 above)
Comment (BoD) [reply build-logic/src/main/kotlin/api.kt]: No you're totally right :) Changed it to a single `version` parameter

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**